tool_path_planner/src/utilities.cpp

```diff
@@ -701,6 +705,133 @@ ToolPaths splitByAxes(const ToolPaths& tool_paths, const Eigen::Vector3f& axis_1
+        std::size_t min_dist = std::min(forward_dist, backward_dist);
+        jumps.push_back(min_dist);
+        if (min_dist > results.largest_jump)
+        {
+          results.largest_jump = min_dist;
+          results.start_i = i;
+          results.end_i = j;
+        }
+      }
+    }
+    if (jumps.size() > 0)
+    {
+      std::sort(jumps.begin(), jumps.end());
+      bool skip_exists = false;
+      for (std::size_t j = 1; j < jumps.size(); j++)
+      {
+        if ((jumps[j] - jumps[j - 1]) > 1)
+        {
+          results.exists = true;
+          skip_exists = true;
+          break;
+        }
+      }
+      results.jump_exists.push_back(skip_exists);
+    }
+  }
+  return results;
+}
+
+std::vector<std::size_t> getSideWithLowerSkipPercentage(largestJumpResults jump_results)
+{
+  double count_inner_skips = 0;
+  double count_inner_total = 0;
+  double count_outer_skips = 0;
+  double count_outer_total = 0;
+  for (std::size_t i = 0; i < jump_results.jump_exists.size(); i++)
+  {
+    if ((i < jump_results.start_i && i < jump_results.end_i) || (i > jump_results.start_i && i > jump_results.end_i))
+    {
+      if (jump_results.jump_exists[i])
+        count_outer_skips++;
+      count_outer_total++;
+    }
+    else if ((i > jump_results.start_i && i < jump_results.end_i) ||
+             (i < jump_results.start_i && i > jump_results.end_i))
+    {
+      if (jump_results.jump_exists[i])
+        count_inner_skips++;
+      count_inner_total++;
+    }
+  }
+  double percent_inner = count_inner_skips / count_inner_total;
+  double percent_outer = count_outer_skips / count_outer_total;
+
+  std::vector<std::size_t> indices_to_keep;
+  if (count_outer_total > count_inner_total)
+  {
+    for (std::size_t i = 0; i <= jump_results.start_i; i++)
+      indices_to_keep.push_back(i);
+    for (std::size_t i = jump_results.end_i; i < jump_results.jump_exists.size(); i++)
+      indices_to_keep.push_back(i);
+  }
+  else
+  {
+    for (std::size_t i = jump_results.start_i; i <= jump_results.end_i; i++)
+      indices_to_keep.push_back(i);
+  }
+  return indices_to_keep;
+}
+
+ToolPathSegment checkForPeninsulaOrInlet(const ToolPathSegment& tool_path_seg,
+                                         const double& min_allowed_width)
+{
+  ToolPathSegment output_segment = tool_path_seg;
+  bool jumps_exist = true;
+  std::size_t jumps_cleared = 0;
+  while (jumps_exist)
+  {
+    largestJumpResults jump_results = getLargestJump(output_segment, min_allowed_width);
+    if (jump_results.exists)
+    {
+      jumps_cleared++;
+      std::vector<std::size_t> indices_to_keep = getSideWithLowerSkipPercentage(jump_results);
+      ToolPathSegment temp_output_segment;
+      for (auto i : indices_to_keep)
+        temp_output_segment.push_back(output_segment[i]);
+      output_segment.clear();
+      output_segment = temp_output_segment;
+    }
+    else
+    {
+      jumps_exist = false;
+    }
+    if (jumps_cleared > 100)
```

Comment:
Is 100 here a magic number?  Or does it have some discrete meaning?

Reply:
It's not a magic number, it is just to catch scenarios that are in an infinite loop. It was really for debugging and should probably be removed or replaced with a check that the path size is consistently decreasing or at zero

Reply:
Is there a more intrinsically meaningful number we could use, such as `initial_waypoints * 2` or even `initial_waypoints - 1`?  (The latter is definitely an upper limit, the lower is probably a theoretical maximum.)